java/jdbc/src/main/java/io/vitess/jdbc/VitessStatement.java

```diff
@@ -643,6 +649,10 @@ protected void checkSQLNullOrEmpty(String sql) throws SQLException {
         return updateCounts;
     }
 
+    private boolean sqlIsUpsert(String sql) {
+        return StringUtils.indexOfIgnoreCase(0, sql, "ON DUPLICATE KEY UPDATE", "\"'`", "\"'`") != -1;
```

Comment:
Is extra white space stripped before this is executed?

Will it handle additional white space between the words?

Reply:
I'm not sure if you are trying to represent two concerns here or one. The mysql connector-j has a better indexOfIgnoreCase that takes a String[] as it's `searchFor`. I'm not sure quite why ours doesn't have that instead. But that would, presumably, handle extra whitespace. I think this already does, since we're running it in a searchMode that include SKIP_WHITE_SPACE.